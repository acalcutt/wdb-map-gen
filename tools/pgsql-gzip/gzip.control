default_version = '1.0'
module_pathname = '$libdir/gzip'
relocatable = true
comment = 'gzip and gunzip functions.'
