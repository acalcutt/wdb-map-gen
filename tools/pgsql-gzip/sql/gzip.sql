CREATE EXTENSION gzip;

SELECT encode(gunzip(gzip('this is my this is my this is my this is my text'::bytea)), 'escape') AS gzip_bytea;
SELECT encode(gunzip(gzip('this is my this is my this is my this is my text'::text)), 'escape') AS gzip_text;
SELECT encode(gunzip(gzip('this is my this is my this is my this is my text')), 'escape') AS gzip_roundtrip;
SELECT gunzip(gzip('\x00000000000000000000'::bytea)) AS gzip_roundtrip_zero;

WITH str AS (
	SELECT repeat('this is my ', 10000) AS str
)
SELECT encode(gunzip(gzip(str)), 'escape') = str AS gzip_long FROM str;

WITH strs AS (
	SELECT repeat('pack my box with five dozen liquor jugs ', generate_series(0, 1000)) AS str
)
SELECT sum((str = encode(gunzip(gzip(str)), 'escape'))::integer) AS gzip_sizes
FROM strs;

SELECT gzip(NULL) AS gzip_null;
SELECT gunzip(gzip('')) AS gzip_empty;
SELECT gunzip(gzip('\x00'::bytea)) AS gzip_zero;

SELECT gunzip(NULL) AS gunzip_null;
SELECT gunzip('') AS gunzip_empty;
SELECT gunzip('\x00'::bytea) AS gunzip_zero;
SELECT gunzip('\x0000'::bytea) AS gunzip_zerozero;
SELECT gunzip('not a gzip bytea'::bytea) AS gunzip_invalid;

