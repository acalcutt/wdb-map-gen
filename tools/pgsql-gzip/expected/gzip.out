CREATE EXTENSION gzip;
SELECT encode(gunzip(gzip('this is my this is my this is my this is my text'::bytea)), 'escape') AS gzip_bytea;
                    gzip_bytea                    
--------------------------------------------------
 this is my this is my this is my this is my text
(1 row)

SELECT encode(gunzip(gzip('this is my this is my this is my this is my text'::text)), 'escape') AS gzip_text;
                    gzip_text                     
--------------------------------------------------
 this is my this is my this is my this is my text
(1 row)

SELECT encode(gunzip(gzip('this is my this is my this is my this is my text')), 'escape') AS gzip_roundtrip;
                  gzip_roundtrip                  
--------------------------------------------------
 this is my this is my this is my this is my text
(1 row)

SELECT gunzip(gzip('\x00000000000000000000'::bytea)) AS gzip_roundtrip_zero;
  gzip_roundtrip_zero   
------------------------
 \x00000000000000000000
(1 row)

WITH str AS (
	SELECT repeat('this is my ', 10000) AS str
)
SELECT encode(gunzip(gzip(str)), 'escape') = str AS gzip_long FROM str;
 gzip_long 
-----------
 t
(1 row)

WITH strs AS (
	SELECT repeat('pack my box with five dozen liquor jugs ', generate_series(0, 1000)) AS str
)
SELECT sum((str = encode(gunzip(gzip(str)), 'escape'))::integer) AS gzip_sizes
FROM strs;
 gzip_sizes 
------------
       1001
(1 row)

SELECT gzip(NULL) AS gzip_null;
 gzip_null 
-----------
 
(1 row)

SELECT gunzip(gzip('')) AS gzip_empty;
 gzip_empty 
------------
 \x
(1 row)

SELECT gunzip(gzip('\x00'::bytea)) AS gzip_zero;
 gzip_zero 
-----------
 \x00
(1 row)

SELECT gunzip(NULL) AS gunzip_null;
 gunzip_null 
-------------
 
(1 row)

SELECT gunzip('') AS gunzip_empty;
ERROR:  decompression error: 
SELECT gunzip('\x00'::bytea) AS gunzip_zero;
ERROR:  decompression error: 
SELECT gunzip('\x0000'::bytea) AS gunzip_zerozero;
ERROR:  decompression error: unknown compression method
SELECT gunzip('not a gzip bytea'::bytea) AS gunzip_invalid;
ERROR:  decompression error: incorrect header check
